jni/src/main/java/com/horizen/merkletreenative/InMemoryLazySparseMerkleTree.java

```diff
@@ -0,0 +1,84 @@
+package com.horizen.merkletreenative;
+
+import java.util.List;
+
+import com.horizen.librustsidechains.FieldElement;
+import com.horizen.librustsidechains.Library;
+
+public class InMemoryLazySparseMerkleTree implements AutoCloseable {
+
+    private long merkleTreePointer;
+
+    static {
+        Library.load();
+    }
+
+    private InMemoryLazySparseMerkleTree(long merkleTreePointer) {
+        if (merkleTreePointer == 0)
+            throw new IllegalArgumentException("merkleTreePointer must be not null.");
+        this.merkleTreePointer = merkleTreePointer;
+    }
+
+    private static native InMemoryLazySparseMerkleTree nativeInit(int height);
+
+    public static InMemoryLazySparseMerkleTree init(int height) {
+        return nativeInit(height);
+    }
+
+    private native boolean nativeIsPositionEmpty(long position);
+
+    public boolean isPositionEmpty(long position){
+        if (merkleTreePointer == 0)
+            throw new IllegalStateException("InMemoryLazySparseMerkleTree instance was freed.");
+        return nativeIsPositionEmpty(position);
+    }
+
+    private native FieldElement nativeAddLeaves(PositionLeaf[] leaves);
+
+    // Add leaves to tree, compute and return the root
+    public FieldElement addLeaves(List<PositionLeaf> leaves){
```

Comment:
Why we should return root?
I would expect to add leaves and return boolean if it was successful